tests/components/netatmo/test_init.py

```diff
@@ -0,0 +1,172 @@
+"""The tests for Netatmo device triggers."""
+from time import time
+from unittest.mock import Mock, patch
+
+import jwt
+
+from homeassistant import config_entries
+from homeassistant.components.netatmo import DOMAIN
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.setup import async_setup_component
+
+from .common import fake_post_request
+
+from tests.common import MockConfigEntry
+from tests.components.cloud import mock_cloud
+
+
+async def test_setup_component(hass):
+    """Test setup and teardown of the netatmo component."""
+    config_entry = MockConfigEntry(
+        domain="netatmo",
+        data={
+            "auth_implementation": "cloud",
+            "token": {
+                "refresh_token": "mock-refresh-token",
+                "access_token": "mock-access-token",
+                "type": "Bearer",
+                "expires_in": 60,
+                "expires_at": time() + 1000,
+                "scope": "read_station",
+            },
+        },
+    )
+    config_entry.add_to_hass(hass)
+
+    with patch(
+        "homeassistant.components.netatmo.api.ConfigEntryNetatmoAuth"
+    ) as mock_auth, patch(
+        "homeassistant.helpers.config_entry_oauth2_flow.async_get_config_entry_implementation",
+    ) as mock_impl, patch(
+        "homeassistant.components.webhook.async_generate_url"
+    ) as mock_webhook:
+        mock_auth.return_value.post_request.side_effect = fake_post_request
+        assert await async_setup_component(hass, "netatmo", {})
+
+    await hass.async_block_till_done()
+
+    assert mock_auth.call_count == 1
+    assert mock_impl.call_count == 1
+    assert mock_webhook.call_count == 1
+
+    assert config_entry.state == config_entries.ENTRY_STATE_LOADED
+    assert len(hass.config_entries.async_entries(DOMAIN)) == 1
+
+    for config_entry in hass.config_entries.async_entries("netatmo"):
+        await hass.config_entries.async_remove(config_entry.entry_id)
```

Comment:
Should the outcome of this be tested, or it's enough that it doesn't throw?

Reply:
Yes